projects/eslint-plugin/rules/strict-tui-doc-example.js

```diff
@@ -0,0 +1,95 @@
+const getTypeName = require('../utils/get-type-name');
+
+const MESSAGE_ID = `strict-doc-example-extensions`;
+const ERROR_MESSAGE = `The import path extension should match the object key`;
+
+const DOC_EXAMPLE_INTERFACE_NAME = 'TuiDocExample';
+
+/**
+ * @type {TuiDocExample}
+ */
+const fileNameToExtension = {
+    TypeScript: '.ts',
+    HTML: '.html',
+    LESS: '.less',
+};
+
+/**
+ * Parses path into RegExp groups.
+ * @param {string} path
+ * @returns {RegExpMatchArray | null}
+ */
+const getPathGroups = path => {
+    const pathRegex = /(.+)(\.(?:ts|less|scss|js|html))(.*)/;
+    return path.match(pathRegex);
+};
+
+/**
+ * @type {import('eslint').Rule.RuleModule}
+ */
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {description: ERROR_MESSAGE},
+        messages: {
+            [MESSAGE_ID]: ERROR_MESSAGE,
+        },
+        fixable: 'code',
+    },
+    create(context) {
+        return {
+            /**
+             * It would be better to support any object expression, not only within a class,
+             * but it's pretty complicated to extract the object type since all parents have different interfaces.
+             */
+            'PropertyDefinition[value.type="ObjectExpression"]'(node) {
+                if (getTypeName(node) !== DOC_EXAMPLE_INTERFACE_NAME) {
+                    return;
+                }
+
+                node.value.properties.forEach(prop => {
+                    // it can be either Identifier or Literal.
+                    const objKey = prop.key.name || prop.key.value;
+                    const objValue = prop.value;
+
+                    if (objValue.type !== 'ImportExpression') {
+                        return;
+                    }
+
+                    const {source} = objValue;
+
+                    /**
+                     * Try to extract an extension from the object key.
+                     * It can be a file name, module path or null.
+                     * @see TuiDocExample
+                     */
+                    const expectedExtension =
+                        fileNameToExtension[objKey] || getPathGroups(objKey)?.[2];
+
+                    /**
+                     * Try to extract an extension from the object value (actually import value).
+                     * It can be a file name or null.
+                     */
+                    const actualPathGroups = getPathGroups(source.value);
+
+                    if (
+                        !(expectedExtension && actualPathGroups) ||
+                        expectedExtension !== actualPathGroups[2]
```

Comment:
```ts
const  mismatchExtensions = !(expectedExtension && actualPathGroups) || expectedExtension !== actualPathGroups[2];

if (mismatchExtensions) {

}
``

Reply:
Also, could you please use named variables
`some1, some2, ..` you can name it more meaningfully

```ts
const groups = actualPathGroups ?? [];
const [, some1, some2, some3] = groups;
// ...
const  mismatchExtensions = !(expectedExtension && groups.length) || expectedExtension !== some2;
// ...
const fixedValue = `'${some1}${expectedExtension}${some3}'`;
```

Reply:
ye, sure, thanks